simpleeval.py

```diff
@@ -238,11 +239,28 @@ def safe_add(a, b):  # pylint: disable=invalid-name
     return a + b
 
 
+def safe_rshift(a, b):  # pylint: disable=invalid-name
+    """ rshift, but with the maximum """
+    if abs(b) > MAX_SHIFT:
```

Comment:
1. Is there any sense in it?
2. Why are such variables global instead of per-instance?
3. The values of them looks very arbitrary. Again, is there any sense in doing this?

Reply:
They are pretty arbitrary, tbh. Essentially I've run some benchmarks on my MacBook, and when it leaps from "close enough to instant" to "annoyingly slow" I'll use something around there.

Yes, I totally agree they should be per instance.

These are things to absolutely get right in 2.x - I didn't want to change the behaviour massively compared to the other current ones.

Thank you so much for your feedback! I really appreciate it!

Reply:
Whether there is sense to it, yes, I think so...  The DOS scenario by too large numbers and calculation is a real concern for anything that allows users entering their own expressions.

How that should be best mitigated is a whole big question..

In some ways the ideal would be to call the calculation in a separate process / thread, if possible sandboxed by the OS, with some kind of timeout enforced by the system.

How much of that is actually possible is another question.

Figuring out a good "evaluate this in another thread with a timeout" may be possible? I don't know if GIL issues may crop up there too?

Reply:
>Whether there is sense to it, yes, I think so... The DOS scenario by too large numbers and calculation is a real concern

What prevents an adversary to structure DoS bit shift into multiple ops, not

```python
a << (MAX_SHIFT * 2 -2)
```

but 

```python
((a << (MAX_SHIFT - 1)) << (MAX_SHIFT - 1))
```

instead?

Reply:
Good point! @KOLANICH 

Hmm...

What do you think?  Should there then be a max on the left side of the operator as well?

(This is leaving aside the problem of how should such limits be calculated for another time...)

Reply:
I guess, limit total size of the result with 64 bits.